segmentation/include/pcl/segmentation/impl/lccp_segmentation.hpp

```diff
@@ -139,8 +139,7 @@ pcl::LCCPSegmentation<PointT>::computeSegmentAdjacency ()
   seg_label_to_neighbor_set_map_.clear ();
 
   //The vertices in the supervoxel adjacency list are the supervoxel centroids
-  std::pair<VertexIterator, VertexIterator> vertex_iterator_range;
-  vertex_iterator_range = boost::vertices (sv_adjacency_list_);
+  std::pair<VertexIterator, VertexIterator> vertex_iterator_range = boost::vertices (sv_adjacency_list_);
```

Comment:
Open question. Which version is better?
```cpp
  std::pair<VertexIterator, VertexIterator> vertex_iterator_range = boost::vertices (sv_adjacency_list_);
  for (VertexIterator sv_itr = vertex_iterator_range.first; sv_itr != vertex_iterator_range.second; ++sv_itr)...
```
or
```cpp
  VertexIterator sv_itr, sv_end;
  for (std::tie(sv_itr, sv_end) = boost::vertices (sv_adjacency_list_); sv_itr != sv_end; ++sv_itr) ...
```

Reply:
I have a slight preference towards the second one because it will be easier to locate it and convert to structural bindings once we switch to c++17.

Reply:
Can't we use range-for loops?

Reply:
Because the iterators don't present a `.begin()` and `.end()` function

Reply:
Got it. There are workarounds available, but better to just use structural bindings later.

In the meantime, should I change it to `std::tie`?

Reply:
`boost::tie` v/s `std::tie`?
```
  EdgeIterator edge_itr, edge_itr_end, next_edge;
  boost::tie (edge_itr, edge_itr_end) = boost::edges (adjacency_list_arg);
```

Reply:
`std::tie` please. There's no reason to use `next_edge` in other cases unless we invalidate the `edge_itr` while in the loop.